common/finalizer/finalizer.go

```diff
@@ -103,7 +107,7 @@ func (f *Finalizer) Deregister(
 // It can only be invoked once; calling it again has no effect.
 // Returns the number of completed callbacks.
 func (f *Finalizer) Run(
-	pool *goro.AdaptivePool,
+	poolFactory func() *goro.AdaptivePool,
```

Comment:
this seems overly complicated. why do we need a factory? can't we just create the pool directly?

Reply:
Since the finalizer is in `common` and the caller might have different ideas about the pool parameters, I don't think the Finalizer itself should now about these details.

Reply:
Or if you're asking why create it lazily, this will save us from spawning goroutines that don't end up being used.

Reply:
there's only one caller and it's using fixed parameters. maybe leave out the flexibility until someone actually wants to use it?

Reply:
Okay